GLOSSARY.md

```diff
@@ -0,0 +1,25 @@
+The SOCI project introduces several new terms that sometimes have subtle differences between them. This glossary defines these terms.
+
+## Terminology 
+
+__SOCI__: Seekable OCI (pronounced so-CHEE). SOCI combines an unmodified [OCI Image](https://github.com/opencontainers/image-spec/blob/v1.0.2/spec.md) (or Docker v2 image) with a SOCI index to enable the SOCI snapshotter to lazily pull the image at runtime.
+
+__SOCI Index__: The combined set of metadata that enables lazy loading of an unmodified OCI image. Conceptually, the SOCI Index is the mapping between files and their locations within a compressed image. In practice, a SOCI Index is made up of a SOCI Index Manifest and zero or more zTOCs. Within a SOCI Index there may be zero or one zTOCs per image layer.
+
+__SOCI Index Manifest__: An [ORAS manfest](https://github.com/oras-project/artifacts-spec/blob/v1.0.0-rc.2/artifact-manifest.md) (soon to be an [OCI Reference Types manifest](https://github.com/opencontainers/wg-reference-types/blob/256c257cc8b725fd324722ee40ead6925b1c8ad8/docs/proposals/PROPOSAL_E.md)) containing the list of zTOCs in the SOCI Index as well as a reference to the image for which the manifest was generated. 
+
+__zTOC__: A Table of Contents for compressed data. A zTOC contains a set of spans of data, filesystem metadata, and locations of files within spans. In SOCI, zTOCs are many-to-one with image layers, that is, there may be many different zTOCs for a single image layer.
```

Comment:
zTOCs don't contain spans. Spans are lazily-loaded from remote. zTOC contains dictionaries, where each dictionary contains the data the individually decompress spans.

Reply:
Yeah, I was being a bit lazy with spans. I used the term "span metadata" when describing how we use zTOCs to extract files. Would that be clear enough here?

I should also highlight that we pull individual spans in the example below as well.

Reply:
I'm fine with "span metadata". If I'm being pedantic, "span metadata" and "dictionary" would be different terms. But maybe that distinction should be part of separate "Deep Dive" document.

Reply:
I would explicitly call out that the zTOC is composed of two separate parts:
1. A table of contents containing file metadata and its offset in the tar file. (That the "TOC".)
2. A collection of "snapshots" of the decompression dictionary at various points in the gzip file. (That's the "z".

Reply:
Decomposing the z and TOC is a good suggestion.

I was trying to avoid getting in to a technical specification and keep this at the conceptual level. In particular, I don't think gzip or dictionaries have a place in the *definition* of a zTOC. Gzip is definitely not required and while my understanding of the theory of compression is pretty surface-level, I don't think dictionary coders are either.

Reply:
1/ I was thinking that for clarity it was okay to be gzip-centric for now. That is, more concrete language would make it easier for the reader to understand what's going on. When and if we add zstd support, we could revise this definition.

But if you feel differently, that's fine too. How about something like:
> A collection of "snapshots" of the state of the compression engine at various points in the layer.

2/ I feel pretty comfortable betting that we will only ever support TAR archives. The only thing that will displace TAR is OCI v2.0. And I feel equally comfortable betting that it will have intrinsic support for lazy-loading, and we can all retire.

Reply:
I would prefer not to be gzip centric here. A doc explaining the technical details of how the snapshotter works can lean on the concreteness of gzip.